app/code/core/Mage/Catalog/Model/Layer.php

```diff
@@ -202,25 +202,34 @@ public function getCurrentStore()
     /**
      * Get collection of all filterable attributes for layer products set
      *
-     * @return Mage_Catalog_Model_Resource_Product_Attribute_Collection|array
+     * @return array
```

Comment:
This is potentially a breaking change, isnt it?

Reply:
Yes, but when you look at where/how it is used I except it to not cause issues. But I could also create a collection and fill the data by hand if needed to enhance compatibility, but I doubt its necessary.